projects/cdk/services/theme-color.service.ts

```diff
@@ -0,0 +1,26 @@
+import {DOCUMENT} from '@angular/common';
+import {inject, Injectable} from '@angular/core';
+import {Meta} from '@angular/platform-browser';
+import {tuiCreateToken} from '@taiga-ui/cdk/utils';
+
+export const TUI_THEME_COLOR = tuiCreateToken<string | null>(null);
+
+@Injectable({
+    providedIn: 'root',
+})
+export class TuiThemeColorService {
+    private readonly meta = inject(Meta);
+    private readonly doc = inject(DOCUMENT);
+    private readonly initial = inject(TUI_THEME_COLOR);
+
+    constructor() {
+        if (this.initial) {
+            this.color = this.initial;
+        }
+    }
+
+    public set color(color: string) {
```

Comment:
There's no way to get current value now. What I wanted was to rename `initial` to `current`, update it in the setter and return in the getter.

Reply:
fixed